src/Teambuilder/client.cpp

```diff
@@ -2055,7 +2057,11 @@ void Client::seeChallenge(const ChallengeInfo &c)
     int challengeId = freeChallengeId();
     if (playerExist(c))
     {
-        if (!call("beforeChallengeReceived(int,int,QString,int)", challengeId, c.opponent(), c.desttier, static_cast<int>(c.clauses))) {
+        if (mychallenges.size() > 4) {
```

Comment:
Instead of adding a good 4 lines of duplicated code (twice below), the decline challenge code could be made into a lambda function. Or it could just be added to the **already existing** `busy()` function:

```cpp
bool Client::busy() const
{
    return away() || mychallenges.size() >= 5;
}
```